dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.test.ts

```diff
@@ -81,99 +40,143 @@ describe('calculateFixedFeeAdjustment', () => {
 
   it('throws a NotFoundError when the found tfm facility does not contain a defined facility snapshot or amendment cover end date', async () => {
     // Arrange
+    const facilityId = '12345678';
+
     when(findOneByUkefFacilityIdSpy)
       .calledWith(facilityId)
       .mockResolvedValue({
-        ...aValidTfmFacility(),
+        ...aTfmFacility(),
         amendments: [],
         facilitySnapshot: {
-          ...aValidFacilitySnapshot(),
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverStartDate: TODAY.getTime().toString(),
           coverEndDate: null,
         },
       });
 
     // Act / Assert
-    await expect(calculateFixedFee(utilisation, facilityId, reportPeriod)).rejects.toThrow(
+    await expect(calculateFixedFee(0, facilityId, aReportPeriod())).rejects.toThrow(
       new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`),
     );
   });
 
   it('throws a NotFoundError when the found tfm facility does not contain a defined facility snapshot cover start date', async () => {
     // Arrange
+    const facilityId = '12345678';
+
     when(findOneByUkefFacilityIdSpy)
       .calledWith(facilityId)
       .mockResolvedValue({
-        ...aValidTfmFacility(),
+        ...aTfmFacility(),
         facilitySnapshot: {
-          ...aValidFacilitySnapshot(),
+          ...aFacility(),
+          ukefFacilityId: facilityId,
+          interestPercentage: 5,
+          dayCountBasis: 365,
+          coverEndDate: TODAY.getTime().toString(),
           coverStartDate: null,
         },
       });
 
     // Act / Assert
-    await expect(calculateFixedFee(utilisation, facilityId, reportPeriod)).rejects.toThrow(
+    await expect(calculateFixedFee(0, facilityId, aReportPeriod())).rejects.toThrow(
       new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`),
     );
   });
 
   it('calculates the fixed fee using the facility snapshot cover start date if the report period start month starts before the cover start date', async () => {
     // Arrange
-    const coverStartDateAfterReportPeriod = addMonths(reportPeriodDate, 1);
+    const facilityId = '12345678';
```

Comment:
These tests are looking better thank you! I think we can still make a few more improvements:
1. We are testing that the correct calculation happens but it feels almost inadvertent, would be good to detail the calculation, e.g. 'calculates the fixed fee by multiplying the utilisation, interest, ... and dividing by ....' 
2. Also are the tests covering cases with decimal places etc and that rounding happens correctly?
3. Finally the answers we assert against feel pulled slightly out of nowhere, I would recommend leaving a comment above the assert detailing/explaining the calculation of the correct value.

I'm actually tempted now I look at this in this format to suggest we actually have a pulled out method just for the last line of the `calculateFixedFee` method which has the actual calculation that we then unit test for the calculation logic so it is tested separately to all the 'fetching and choosing data' logic... I'm not sure what it would be called though... perhaps the existing method is `getFixedFee` and then the calculation line pulled out would be `calcualteFixedFee`?

Reply:
That's a nice idea - makes it much easier to separate the concerns of getting the correct data and calculating the value